DataFormats/PatCandidates/src/Jet.cc

```diff
@@ -584,8 +592,21 @@ void Jet::cachePFCandidates() const {
   // Set the cache
   pfCandidatesTemp_.set(std::move(pfCandidatesTemp));
 }
-
-
+ 
+/// method to cache the daughters to allow "user-friendly" access
+void Jet::cacheDaughters() const {
+    // Jets in MiniAOD produced via JetSubstructurePacker contain a mixture of subjets and particles as daughters
+    std::unique_ptr<std::vector<reco::CandidatePtr>> daughtersTemp{ new std::vector<reco::CandidatePtr>{}};
+    const std::vector<reco::CandidatePtr> & jdaus = daughterPtrVector();
+    for (const reco::CandidatePtr & dau : jdaus) {
+      if (dau->isJet()) {
+        const std::vector<reco::CandidatePtr> & sjdaus = edm::Ptr<pat::Jet>(dau)->daughterPtrVector();
```

Comment:
I would prefer
````
  const pat::Jet *subjet = dynamic_cast<const pat::Jet *>(&*dau);
  if (subjet) { ... 
````
or perhaps with `reco::Jet` instead of `pat::Jet`?
there's no guarantee that `isJet()` will return true only a `pat::Jet` object (and there may be overheads in doing the cast using edm::Ptr, which is unnecessary since you're not writing it to a file)